src/components/widget/List/RawList.js

```diff
@@ -8,7 +8,26 @@ class RawList extends Component {
 
         this.state = {
             selected: props.selected || 0,
-            isOpen: false
+            isOpen: false,
+            dropdownList: []
+        }
+    }
+
+    componentDidUpdate = (prevProps) => {
+        const { list, mandatory } = this.props;
+
+        if(prevProps.list !== list){
+            let dropdown = [];
+
+            if(!mandatory){
+                dropdown.push(0);
+            }
+
+            if(list.length > 0) {
+                this.setState(Object.assign({}, this.state, {
```

Comment:
@Dunkat We are going away from the assumption, to always provide deep copy of object, event if it is one level deep. `setState` is doing that for us.